cloud/google/gceproviderconfig/types.go

```diff
@@ -41,3 +40,10 @@ type DiskInitializeParams struct {
 	DiskSizeGb int64  `json:"diskSizeGb"`
 	DiskType   string `json:"diskType"`
```

Comment:
Are we sure we want to say the machines in a cluster have to be part of one and only one project? Seems like we are introducing a limitation and it would be good to call out reasons why.

Reply:
The confusing part if we keep the project on the machine is that the cluster controller will need to know which projects the cluster is comprised of. In that case either:

a) The cluster controller needs to watch the machines and update/create/delete cloud resources based on which projects the current machines belong to
b) the cluster controller will have a list of allowed projects and the machines will fail to create if their project isn't in the list

The cloud resources would be things like networks, firewall rules to allow the nodes to access each other, service account permissions on all projects for the machine/ingress controllers. Keeping these up to date with a changing list of projects could add some complication to the cluster controller.

Reply:
Nodes spanning multiple projects is certainly something we **could** do, but I'm not sure I see the reason. If we do that, I would much prefer the approach of having a project list in the cluster and a project field in the machine.

Reply:
Hmmm.... I am cool with keeping is simple and in one project for now. However, could you open an issue to see if there is anyone who has a scenario/would want machines in different projects? That way we can gauge the need to implement it in the future. :)

Reply:
Yea that's a good idea. It seems like something we could change pretty easily later. I don't think we're add a lot more work by not doing it this way in the beginning.